### DISCUSSION TITLE
Which roles do we really need?

### DISCUSSION THREAD

**Discussion by dpi209** [CosmoQuestX/CSB7.0#137] [Future design]
After a recent hiccup with the cosmoquest sandbox, I've been thinking about the current role structure. Here's some of my thoughts:

* We currently have a superadmin and an admin defined, the difference being that the superadmin can create admins and the admin can't. Also we have a role currently called "moderator" which can probably do less than an admin but isn't currently clearly defined? Should we discern further on the site level? Like, for example, have a user admin that can only change user rights and reset passwords or a project admin that can only change project-related stuff but is not allowed to modify users? Should we have a "staff" role that doesn't have more privileges 
* On the project level, we have a project lead, which is basically supposed to be an admin on a project level (can create applications, can give the "scientist" role for their project to users, allowing them to access scientific data for their project. 
* On the user level, there are currently two special users, which are not optimally named. One is the poweruser, which is supposed to be someone you get to betatest your application, and one is the hero, which used to be defined as a proficient mapper. The latter is probably useful if you want to have an application "limited to heroes". I'm not too sure whether the poweruser/betatester would see much use. 
* Lastly, If every user gets the "standard user" role, can't we just assume the presence of the role and only enter special roles into the database?

What do you think?

**Comment by starstryder** [comment]
I'm working on defining things here: https://github.com/CosmoQuestX/CSB7.0/wiki/User-Management

The standard user is like on Discord where folks have to agree to site rules before they can see shit. I'm going to add a feature request to create a "Agree to terms"

**Reply by dpi209** [reply]
That makes a lot of sense!
Any thoughts about moderator/staff and poweruser/hero?
